### ISSUE TITLE
Child nodes are cloned rather than updated when creating a container node

### ISSUE THREAD

**Issue by Goodwine** [postcss/postcss#1987]
Example:
```js
const decl = postcss.decl({ prop: 'foo', value: 'bar' });
const atRule = postcss.atRule({
  name: 'foo',
  nodes: [decl],
});
const comment = postcss.comment({ text: 'hello' });
// TypeError: Cannot read properties of undefined (reading 'insertBefore')
decl.before(comment);
```

Is this intended? I'd expect the child nodes to be updated, but instead they get cloned, and therefore the original instance is not updated.

**Comment by ai** [comment]
Here is related commit https://github.com/postcss/postcss/commit/9d19bce33a43f6964a868776d8dc53f7ada8e209 with test of the case why we needed it

We added `clone()` for the case when you move nodes from one root to another.

Maybe we can add some check to separate your root-less case (because we don’t need clone here). But I will need a PR since it is not critical.

**Comment by Goodwine** [comment]
Ah, do you mean adding a check that if `node.root() == null`, or maybe even just checking whether `node.parent == null`, then don't clone the node?

I can work on a PR to do that. Do you think it'll be a good idea? Although I'd like this feature, I wonder if the different behavior (to clone, or not to clone) may be confusing to the user. I guess it's fine as long as it's documented that the child nodes will always be cloned except when their parent is null? 🤔 

If you can confirm this is desirable, then I'll go ahead and work on a PR. But if you think it's not desirable to have different behaviors, that also makes sense to me and we can close this issue.

**Comment by ai** [comment]
> Ah, do you mean adding a check that if node.root() == null, or maybe even just checking whether node.parent == null, then don't clone the node?

Yes. But exact check should be found by some small investigation (I don’t have answer)

**Comment by ai** [comment]
> Although I'd like this feature, I wonder if the different behavior (to clone, or not to clone) may be confusing to the user. 

It is OK because this clone feature was introduced as a fix. So it is hidden behavior already.
